File: greedy/cpp/455.assign-cookies.cpp
/*
 * @lc app=leetcode id=455 lang=cpp
 *
 * [455] Assign Cookies
 */

// @lc code=start
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int child_idx = 0, cookie_idx = 0;
        int n_children = g.size(), n_cookies = s.size();
        while (child_idx < n_children && cookie_idx < n_cookies) {
            if (s[cookie_idx] >= g[child_idx]) {
                child_idx += 1;
            }
            cookie_idx += 1;
        }

        return child_idx;
    }
};
// @lc code=end

File: greedy/swift/455.assign-cookies.swift
/*
 * @lc app=leetcode id=455 lang=swift
 *
 * [455] Assign Cookies
 */

// @lc code=start
class Solution {
    func findContentChildren(_ g: [Int], _ s: [Int]) -> Int {
        var children_array = g.sorted(by: <)
        var cookies_array = s.sorted(by: <)

        print(children_array)
        print(cookies_array)

        var children_index = 0
        var cookie_index = 0

        while children_index < children_array.count && cookie_index < cookies_array.count {
            if cookies_array[cookie_index] >= children_array[children_index] {
                children_index += 1
            }
            cookie_index += 1
        }

        return children_index
    }
}
// @lc code=end

